### ISSUE TITLE
`:set -package ...` not supported yet

### ISSUE THREAD

**Issue by teto** [gibiansky/IHaskell#1210]
I would like to load  some custom libraries in ihaskell. I've tried to load them with `:set -package-db ` which returns a `Warning: -package not supported yet` stemming from
https://github.com/gibiansky/IHaskell/blob/5b7e03b2caad17a51cb7490d66bf808e0e8b9d4a/src/IHaskell/Eval/Util.hs#L219

Any hope it can be supported or a workaorund ? Our libraries are built by bazel if that's of any help (so not easily convertible to nix packages) and we are using ihaskell through tweag/jupyterWith.

A previously similar issue https://github.com/gibiansky/IHaskell/issues/692 . Apparently you use the GHC api directly, hie-bios does pass all the flags so that should be possible.

**Comment by vaibhavsagar** [comment]
It looks like the relevant section of the GHCi source is here: https://github.com/ghc/ghc/blob/11bdf3cdd6efb406839a0ebe33455908a66df805/ghc/GHCi/UI.hs#L3094-L3108. If you'd like to adapt it to IHaskell I would be happy to review and merge.

**Comment by teto** [comment]
I've looked at ghc implementation for this and this looks non trivial as few symbols are exported and it would be hard to maintain this for different ghc implementations.
I would like to allow ihaskell to find the existing libraries instead (which is non-trivial because I have a mix of ghc and bazel-built libraries). 
So my question is how does ihaskell find the libraries ? Does it rely on `Ihaskell  --ghclib=<path>` exclusively ? or can I configure a ghc with different package databases ? The latter would be easier for me since my libraries are scattered across different folders.

**Comment by vaibhavsagar** [comment]
It primarily uses `--ghclib` but I recently added support for package environment files in https://github.com/gibiansky/IHaskell/pull/1280, so you can use multiple package databases that way if you need. I chose not to add that support for GHC < 9.0 since nobody asked for it and if you want to do the same thing with `:set -package` that would be fine by me.

**Comment by teto** [comment]
is that the file referenced by GHC_ENVIRONMENT ? that's super cool. I will run some tests with ghc8 in https://github.com/teto/IHaskell/pull/new/ghc8-env (will open PR if that's conclusive).

**Comment by vaibhavsagar** [comment]
Yes, but I meant it for use with `cabal`, e.g. https://github.com/gibiansky/IHaskell/issues/1012#issuecomment-932889297.
